aws/services/VPC/VPC_With_Managed_NAT_And_Private_Subnet.yaml

```diff
@@ -1,15 +1,57 @@
 ---
   AWSTemplateFormatVersion: "2010-09-09"
+
   Description: "Creates a VPC with Managed NAT, similar to the VPC Wizard at https://console.aws.amazon.com/vpc/home#wizardFullpagePublicAndPrivate: (extended from VPC_with_PublicIPs_And_DNS.template sample)"
+
   Mappings:
+
     SubnetConfig:
       VPC:
         CIDR: "10.0.0.0/16"
-      Public:
+      Public0:
         CIDR: "10.0.0.0/24"
-      Private:
+      Public1:
         CIDR: "10.0.1.0/24"
+      Private0:
+        CIDR: "10.0.2.0/24"
+      Private1:
+        CIDR: "10.0.3.0/24"
+
+    # This might seem repetitive but there are occasions when
```

Comment:
This is an interesting approach to the AZ problem. Is the intent that if an and user is blocked from a specific AZ they can just update the map?  Might want to doc that a little more specifically.

Reply:
Yep, that's exactly it. This caused me untold problems in my previous job, the `b` AZ was not available to me to create subnets in `ap-northeast-1` and I had to litter my VPC template with a bunch of conditionals based on region to work around it; this seems to me to be to be the simplest solution to mitigate that. I can update the comment to be explicit about this.

Reply:
*done